modules/mod-deep-learning/ModelCard.h

```diff
@@ -0,0 +1,212 @@
+/**********************************************************************
+
+   Audacity: A Digital Audio Editor
+   Audacity(R) is copyright (c) 1999-2021 Audacity Team.
+
+   ModelCard.h
+   Hugo Flores Garcia
+
+******************************************************************/
+/**
+
+\class ModelCard
+\brief model metadata for deep learning models
+
+*/
+/*******************************************************************/
+
+#pragma once
+
+#include <torch/script.h>
+#include <torch/torch.h>
+
+#include <rapidjson/document.h>
+#include <rapidjson/schema.h>
+#include "rapidjson/prettywriter.h" 
+#include <rapidjson/writer.h>
+#include <wx/string.h>
+#include <wx/log.h>
+
+#include "MemoryX.h"
+#include "AudacityException.h"
+
+using Doc = rapidjson::Document;
+
+// this exception should be caught internally, but we 
+// derive from MessageBoxException just in case it needs to 
+// get handled by Audacity
+class InvalidModelCardDocument final : public MessageBoxException
+{
+public:
+   InvalidModelCardDocument(const TranslatableString msg, std::string trace): 
+                           m_msg(msg), 
```

Comment:
You can `std::move` from `msg` too for a small efficiency

Reply:
I looked at all uses of this type, and I don't see a non-empty second argument given anywhere to the constructor.

Reply:
resolved in  fafafa2af